### ISSUE TITLE
Add option to allow Cross origin request

### ISSUE THREAD

**Issue by zyfdegh** [yudai/gotty#108]
I have a use case to call gotty cross origin. It will encounter 403 forbidden error.  I'm a new backend golang developer, and don't know much about http protocol. Can CORS enabled with an option?

**Comment by zyfdegh** [comment]
Well, I solved this problem by fork this repo and changed gorilla/websocket source code under directory `vendor`. In `vendor/github.com/gorilla/websocket/server.go`, I commented this 3 lines:

```go
//	if !checkOrigin(r) {
//		return u.returnError(w, r, http.StatusForbidden, "websocket: origin not allowed")
//	}
```

Now, it works in my case. I'm not sure about if any security issue may occur later.

**Comment by yudai** [comment]
Added a new option `ws-origin` which you can provide a regexp that matches your origin servers.
Thanks for the request.

**Comment by srid** [comment]
@yudai How exactly is the `ws-origin` option supposed to work? Even specifying a catch-all regex doesn't seem to have any effect:

```
$ ~/go/bin/gotty -a 0.0.0.0 -p 9999 --ws-origin "\*" ob run
...
2019/03/26 19:00:27 Connection closed by websocket: origin not allowed: 10.100.0.1:44474, connections: 0/0
```

I ended up patching gotty to [disable the origin checker altogether](https://github.com/srid/mynixos/commit/954ca5e0680c063bf26a1843be4e85efcb00f17d).

**Comment by ingobaab** [comment]
@srid:
You can allow all websocket-origins with `'.*'` i.e.:
`TERM=xterm gotty --ws-origin '.*' htop`
I also would like to see an example limiting ws-origin to one (internal) IPv4 !
